Assets/Scripts/Agents/AgentBase.cs

```diff
@@ -95,6 +95,23 @@ public Transformation GetRelativeTransformation(in Vector3 waypoint) {
     return GetRelativeTransformation(waypoint, Vector3.zero, Vector3.zero);
   }
 
+  private void AlignWithVelocity() {
+    const float speedThreshold = 0.1f;
+    const float rotationSpeedDegreesPerSecond = 10000f;
+
+    // Only align if the velocity is significant.
+    if (Speed > speedThreshold) {
+      // Create a rotation with the forward direction along the velocity vector and the up direction
+      // along world up.
+      Quaternion targetRotation = Quaternion.LookRotation(Velocity, Vector3.up);
+
+      // Smoothly rotate towards the target rotation.
+      transform.rotation = Quaternion.RotateTowards(
+          transform.rotation, targetRotation,
+          maxDegreesDelta: rotationSpeedDegreesPerSecond * Time.fixedDeltaTime);
+    }
```

Comment:
There is still an AlignWithVelocity implementation in Agent.cs. Maybe you're adding this in anticipation of later switching over to using AgentBase::AlignWithVelocity?

Reply:
Yes, Agent.cs will be deleted.